akka-persistence/src/main/scala/akka/persistence/snapshot/SnapshotStore.scala

```diff
@@ -44,10 +44,18 @@ trait SnapshotStore extends Actor {
       delete(metadata)
```

Comment:
As I read if it is valid, flow is:

* save future fails 
* gets converted to SaveSnapshotFailure and piped `to (self, p)` 
* which means sender for `SaveSnapshotFailure` is received, we issue a delete (that line)
* and send the event to the original persistent actor.